### PR TITLE
feat: parse pet acquisition milestone

### LINKED ISSUE #168: Pet: Add completions/XP from Clan chat to Dink notification
**Dink version:** 1.3.0

Since the pet notifications now show up with a name. I wonder if it's possible to add the completion / XP section from the GIM / Clan chat notification to the Dink notification?

What the notification now looks like:
Dink pet notification without KC

The GIM / Clan chat notification:
Notification Pet Youngleff

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #168

image

### CODE REVIEW

File: src/main/java/dinkplugin/notifiers/PetNotifier.java [hunk 1]
```diff
@@ -77,7 +79,7 @@ public void onChatMessage(String chatMessage) {
     }
 
     public void onClanNotification(String message) {
-        if (!PRIMED_NAME.equals(petName)) {
+        if (petName == null) {
```
[the remark this answers is not in the document]
Comment (iProdigy) [reply src/main/java/dinkplugin/notifiers/PetNotifier.java:82]: Your logic is correct; another message may have given the exact pet name but not the milestone, so we run milestone matching in more cases with this looser check. As for a regression test, I believe it is already covered by https://github.com/pajlads/DinkPlugin/blob/eb9ec9f05619a2c4de8b4ac78603e7ee371eb3b0/src/test/java/dinkplugin/notifiers/PetNotifierTest.java#L144-L152

File: src/main/java/dinkplugin/message/Field.java [hunk 2]
```diff
@@ -20,6 +21,10 @@ public class Field {
     @Nullable Boolean inline;
 
     public Field(String name, String value) {
-        this(name, value, null);
+        this(name, value, true);
+    }
+
+    public static String format(String language, String value) {
```
Comment (pajlada) [inline src/main/java/dinkplugin/message/Field.java:27]: Maybe name `formatBlock` or `formatCodeBlock` or `formatValue` as that's how its most often used? and changing the name of the `language` parameter to something like `codeBlockLanguage` to clarify

File: src/main/java/dinkplugin/message/Field.java [hunk 3]
```diff
@@ -20,6 +21,10 @@ public class Field {
     @Nullable Boolean inline;
 
     public Field(String name, String value) {
-        this(name, value, null);
+        this(name, value, true);
+    }
+
+    public static String formatBlock(String codeBlockLanguage, String content) {
```
Comment (iProdigy) [inline src/main/java/dinkplugin/message/Field.java:27]: I'm also open to moving this method to Utils even though it's only used for field value formatting

Comment (pajlada) [reply src/main/java/dinkplugin/message/Field.java:27]: I'm fine with this staying here until we find a use for it outside of Fields

**Review by pajlada — APPROVED** [review]

**Pull request merged**